### PR TITLE
Always allow ignition while attached to launch clamps

### PR DESCRIPTION
**Pull request by ec429**
This allows you to retry if ignition failed, as long as you're still on
the pad (and thus motivates stage-and-a-half designs like Atlas Classic,
or parallel-staged designs like the R-7).
However, ignitions (and ignitionResources) will still be consumed, so if
you have a multi-ignition first stage engine, this won't be what you
want.  I decided not to bother adding extra complications to deal with
this, as it seems a fairly uncommon case (by the time restartable first
stage engines like the Merlin are developed, ignition reliability is
basically 1).

Because one use case for this is making some engines only ignitable on
the pad (ignitions=0), have a special case for this in GetInfo output.

Also, add a flag field to ModuleEngineConfigs which, if set, specifies
that ignitions=0 actually means zero, rather than unlimited.

### CODE REVIEW

**Comment by ec429** [comment]
NB. I have an RO configs patch showing the use of this for the LR105.  See https://github.com/ec429/RealismOverhaul/tree/lr105-s4

**Comment by NathanKell** [comment]
Looks good to me! I do recall discussing this in IRC a ways back.
